internal/bminventory/inventory.go

```diff
@@ -1500,23 +1500,67 @@ func (b *bareMetalInventory) refreshClusterHosts(ctx context.Context, cluster *c
 		log.WithError(err).Errorf("Failed to set cluster %s majority groups", cluster.ID.String())
 		return common.NewApiError(http.StatusInternalServerError, err)
 	}
+
+	dbCluster, err := common.GetClusterFromDB(tx, *cluster.ID, true)
```

Comment:
Please use https://github.com/openshift/assisted-service/blob/master/internal/common/db.go#L68-L73 instead of bool